pandas/io/formats/info.py

```diff
@@ -83,30 +92,32 @@ def _initialize_memory_usage(
 
 
 class BaseInfo(ABC):
-    """Base class for DataFrameInfo and SeriesInfo.
+    """
+    Base class for DataFrameInfo and SeriesInfo.
 
     Parameters
     ----------
-    data : FrameOrSeries
+    data : FrameOrSeriesUnion
```

Comment:
@simonjayhawkins is this right here?

Reply:
The variable type annotation for `data` is correct if this doesn't need to be a Generic class.

FrameOrSeries is a typevar and using a typevar for a variable type annotation gives `error: Type variable "pandas._typing.FrameOrSeries" is unbound  [valid-type]`

However, for the docstring, I think it is still preferred not to use Type notation, even for internal docstrings and follow the documentation contributing guide.

Reply:
Changed that.